Slim/Plugin/Podcast/Plugin.pm

```diff
@@ -69,6 +73,23 @@ sub initPlugin {
 		func   => \&trackInfoMenu,
 	) );
 	
+	foreach ( @{$prefs->get('feeds')} ) {
+		my $url = $_->{value};
+
+		Slim::Networking::SimpleAsyncHTTP->new(
+			sub { 
+				eval {
+					my $xml = XMLin(shift->content);
+					my $image = $xml->{channel}->{image}->{url} || $xml->{channel}->{'itunes:image'}->{href};
```

Comment:
That would crash the eval if we didn't have `$xml->{channel}->{image}`, not even try the iTunes artwork. I hate when I have to do this, but we'll have to verify each element before accessing it. At some point I should see whether something like [lodash's `get()`](https://lodash.com/docs/4.17.15#get) does exist for Perl, or implement it. It's a pattern I have to use so often, I bet others have done this before.

Reply:
One more thing: can this be delayed, or tamed otherwise? In many places we try to not rush all requests required around startup, as we can easily end up with dozens of requests just to get LMS started. I do understand that we'll need those covers in the main menu potentially. But we can live with a delay, can't we? So either serialize the requests using `Slim::Utils::Scheduler`, or put them on a timer with some "random" offset.

Reply:
I can also put these back (I did that a the beginning) to the feed handler so that it its the user only once and I extend cache for 90 days. And anyway, this change does not work after 30 days, so I'll have to reconsider

Reply:
> That would crash the eval if we didn't have `$xml->{channel}->{image}`, not even try the iTunes artwork. I hate when I have to do this, but we'll have to verify each element before accessing it. At some point I should see whether something like [lodash's `get()`](https://lodash.com/docs/4.17.15#get) does exist for Perl, or implement it. It's a pattern I have to use so often, I bet others have done this before.
>
I'm not following. The eval {} protects us from anything happening there? I just tried something like that and it does not carp
``
my $a = 2;
eval {
    my $b = $a->{item}->{data};
    print $b;
}
```

Reply:
> I'm not following. The eval {} protects us from anything happening there? I just tried something like that and it does not croak

Yes. But as the access to `$xml->{channel}->{image}` already would end the execution, the second part of the assignment never even comes into play.

Reply:
Where I'm confused is that such code
```
my $a = { item1 => 2, item2 => { x => 1, data => 10 } };
my $b = $a->{empty}->{empty}->{empty} || $a->{item2}->{data};
print $b;
```
does not croak and prints 10. It only does if $a is not a hash or undef, which in our case is never

Reply:
I'm speechless... I don't know how many time such constructs gave me `Can't use an undefined value as a HASH reference`. But oh well...